block/src/Pos/Block/Network/Logic.hs

```diff
@@ -302,15 +308,16 @@ applyWithRollback
        , HasGenesisData
        , HasGenesisHash
        , HasProtocolConstants
+       , HasProtocolMagic
        , HasGenesisBlockVersionData
+       , HasMisbehaviorMetrics ctx
        )
-    => NodeId
-    -> Diffusion m
+    => Diffusion m
     -> OldestFirst NE Block
     -> HeaderHash
     -> NewestFirst NE Blund
     -> m ()
-applyWithRollback nodeId diffusion toApply lca toRollback = do
```

Comment:
The `nodeId` was only ever needed for reporting a "misbehavior". `reportMisbehavior` never actually did anything though (`reportMisbehavior = when False $ ...`).